core/dbt/task/init.py

```diff
@@ -274,9 +276,18 @@ def run(self):
             in_project = False
 
         if in_project:
+            # If --profile was specified, it means use an existing profile, which is not
+            # applicable to this case
+            if getattr(get_flags(), "PROFILE", None):
+                print(
+                    "Can not init existing project with specified profile, edit dbt_project.yml instead"
+                )
+                sys.exit(1)
+
             # When dbt init is run inside an existing project,
             # just setup the user's profile.
             profile_name = self.get_profile_name_from_current_project()
+            profile_specified = False
```

Comment:
Sentinel values like this in if-else branches can make it hard to read code and track execution semantics. Would much prefer a different strategy like calling functions that make sense in each if-else branch.

Reply:
I personally can handle sentinel values as long as the names are descriptive of the state they capture, and use is not excessive. However, there was a request elsewhere to refactor run() into smaller methods. Since creating the profile is triggered in two places, its an excellent candidate for a refactor. Changed as requested.